benchmarking/util.py

```diff
@@ -0,0 +1,114 @@
+
+    Filter the list of config files to ignore the seed, and only return the last
+    file per algorithm and environment.
+
+    Args:
+        files: list of config files
+
+    Returns:
+        list of config files
+    """
+    config_files = [pathlib.Path(config_file) for config_file in files]
+    config_files = [
+        config_file for config_file in config_files if config_file.name == "config.json"
+    ]
+    # all files have the following format:
+    # [/base/path/to/file]/<algo_and_environment>/<other_info>/sacred/1/config.json
+    experiments = {}
+    for config_file in config_files:
+        experiment = config_file.parents[3]
+        if experiment not in experiments:
+            experiments[experiment] = [config_file]
+        else:
+            experiments[experiment].append(config_file)
+    final_config_files = []
+    for experiment, config_files in experiments.items():
+        config_files.sort(key=lambda config_file: config_file.parents[1])
+        final_config_files.append(config_files[-1])
+    return final_config_files
+
+
+def remove_empty_dicts(d: dict):
+    """Remove empty dictionaries in place.
+
+    This is a recursive function that will remove empty dictionaries from a dictionary.
+
+    Args:
+        d: dictionary to be filtered
+    """
+    for key, value in list(d.items()):
+        if isinstance(value, dict):
+            remove_empty_dicts(value)
+            if not value:
+                d.pop(key)
+        elif value == {}:
+            d.pop(key)
+
+
+def clean_config_file(file: pathlib.Path, write_path: pathlib.Path, /) -> None:
+    """Clean a config file.
+
+    reads the file, loads from json to dict, removes keys related to e.g. seeds,
+    config paths, leaving only hyperparameters, and writes back to file.
+
+    Args:
+        file: path to config file
+        write_path: path to write the cleaned config file
+    """
+    import json
+
+    with open(file) as f:
+        config = json.load(f)
+    # remove key 'agent_path'
+    config.pop("agent_path")
+    config.pop("seed")
+    config.get("demonstrations", {}).pop("rollout_path")
+    config.get("expert", {}).get("loader_kwargs", {}).pop("path", None)
+    env_name = config.pop("common").pop("env_name")
+    config["common"] = {"env_name": env_name}
+    config.pop("show_config", None)
+
+    remove_empty_dicts(config)
+    # files are of the format
+    # /path/to/file/example_<algo>_<env>_best_hp_eval/<other_info>/sacred/1/config.json
+    # we want to write to /<write_path>/<algo>_<env>.json
+    with open(write_path / f"{file.parents[3].name}.json", "w") as f:
+        json.dump(config, f, indent=4)
+
+
+if __name__ == "__main__":
+    # get two arguments from the terminal. The first positional argument contains the
+    # path to a txt file that has a list of paths to config files.
+    # The second positional argument is the path to the directory where
+    # the cleaned config files should be written.
+    import argparse
```

Comment:
Style guide violation: https://google.github.io/styleguide/pyguide.html#313-imports-formatting Imports should always be at top level of module to make it clear what modules file depends on, unless there's an explicit need for them to be conditional/only locally imported.

I'd also suggest moving all this to a `main()` function that's called in the `__main__` guard block to avoid polluting the global namespace.